File: kartwheel/app/src/main/java/us/handstand/kartwheel/model/AndroidStorage.java
package us.handstand.kartwheel.model;


import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.annotation.StringDef;

public class AndroidStorage {
    public static final String USER_ID = "user_id";
    public static final String EMOJI_CODE = "emoji_code";
    public static final String SERVER = "server";
    public static final String PORT = "port";

    @StringDef({USER_ID, EMOJI_CODE, SERVER, PORT})
    public @interface KEYS {
    }

    private static AndroidStorage instance;

    private final Context context;

    private AndroidStorage(Context context) {
        this.context = context.getApplicationContext();
    }

    private SharedPreferences getPrefs() {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }

    public static void initialize(Context context) {
        if (instance == null) {
            instance = new AndroidStorage(context);
        }
    }

    public static String getString(@KEYS String key) {
        return instance.getPrefs().getString(key, null);
    }

    public static int getInt(@KEYS String key) {
        return instance.getPrefs().getInt(key, 80);
    }

    public static void set(@KEYS String key, String value) {
        instance.getPrefs().edit().putString(key, value).apply();
    }

    public static void set(@KEYS String key, int value) {
        instance.getPrefs().edit().putInt(key, value).apply();
    }

    public static String getUserId() {
        return getString(USER_ID);
    }
}

File: kartwheel/app/src/main/kotlin/us/handstand/kartwheel/model/AndroidStorage.kt
package us.handstand.kartwheel.model


import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import android.support.annotation.StringDef

class AndroidStorage private constructor(context: Context) {
    private val context: Context

    init {
        this.context = context.applicationContext
    }

    private val prefs: SharedPreferences
        get() = PreferenceManager.getDefaultSharedPreferences(context)

    companion object {
        const val USER_ID = "user_id"
        const val TEAM_ID = "team_id"
        const val EVENT_ID = "event_id"
        const val EMOJI_CODE = "emoji_code"

        @StringDef(USER_ID, EMOJI_CODE, EVENT_ID, TEAM_ID)
        annotation class KEYS

        private var instance: AndroidStorage? = null

        fun initialize(context: Context) {
            if (instance == null) {
                instance = AndroidStorage(context)
            }
        }

        fun get(@KEYS key: String): String {
            return instance!!.prefs.getString(key, null)
        }

        operator fun set(@KEYS key: String, value: String) {
            instance!!.prefs.edit().putString(key, value).apply()
        }
    }
}
